dune-buggy.js

```diff
@@ -64,49 +68,122 @@ async function fetchTripLogsForVehicle(vehicleLogs, vehicle_id) {
+    .map((logEntry) =>
+      _.get(logEntry, "jsonPayload.response.remainingVehicleJourneySegments")
+    )
+    .flatten()
+    .map((segment) => _.get(segment, "stop.tasks"))
+    .flatten()
+    .map((tasks) => _.get(tasks, "taskId"))
+    .flatten()
+    .uniq()
+    .compact()
+    .value();
+  let taskLogs = [];
+  if (task_ids.length > 0) {
+    console.log("gots task_ids", task_ids);
+    taskLogs = await logging.fetchLogs("task_id", task_ids, argv.daysAgo);
+  } else {
+    console.log(`no tasks associated with vehicle id ${argv.vehicle}`);
+  }
+  return taskLogs;
+}
+
+async function fetchVehicleLogs(vehicle, trip) {
+  const label = vehicle ? "vehicle_id" : "trip_id";
+  const labelVal = vehicle ? vehicle : trip;
+
+  console.log(`Fetching logs for ${label} = ${labelVal}`);
+  return await logging.fetchLogs(label, [labelVal], argv.daysAgo);
+}
+
+async function fetchDeliveryVehicleLogs(deliveryVehicle, vehicleLogs) {
+  if (vehicleLogs.length !== 0) {
+    // regular vehicle logs found, not a deliveryVehicle
+    return [];
+  }
+  // TODO: is it more efficient to run the log query twice
+  // or to update the log filter with an OR?
+  //
+  // Could also force the user to specify which type of vehicle they're interested
+  // in on the command line -- but that seems unfriendly & error prone
+  console.log("fetching logs for deliveryVehicle", deliveryVehicle);
+  return await logging.fetchLogs(
+    "delivery_vehicle_id",
+    [deliveryVehicle],
+    argv.daysAgo
+  );
+}
+
 async function main() {
   if (argv.vehicle) {
     console.log("Fetching logs for vehicle id", argv.vehicle);
-    label = "vehicle_id";
-    labelVal = argv.vehicle;
   } else if (argv.trip) {
     console.log("Fetching logs for trip id", argv.trip);
-    label = "trip_id";
-    labelVal = argv.trip;
   } else if (argv.task) {
     console.log("Not implemented yet: the task id is:", argv.task);
     return;
   } else {
     yargs.showHelp();
     return;
   }
+  // TODO: error if both trip & vehicle being set
 
   await auth.init();
-  let rawLogs = await logging.fetchLogs(label, [labelVal], argv.daysAgo);
-  let tripLogs = [];
-  if (argv.vehicle) {
-    tripLogs = await fetchTripLogsForVehicle(rawLogs, argv.vehicle);
-  }
-  rawLogs = _(rawLogs)
+
+  // TOOD: handle lookup by task -- task logs are trickier in than
+  // updateDeliveryVehicleLogs aren't labeled with the task, since there
+  // are many tasks at any given time(unlike how
+  // relevant updateVehicleLogs are labeled by a trip_id)
+  const vehicleLogs = await fetchVehicleLogs(argv.vehicle, argv.trip);
+  const deliveryVehicleLogs = await fetchDeliveryVehicleLogs(
+    argv.vehicle,
+    vehicleLogs
+  );
+  const taskLogs = await fetchTaskLogsForVehicle(
+    argv.vehicle,
+    deliveryVehicleLogs
+  );
+  const tripLogs = await fetchTripLogsForVehicle(argv.vehicle, vehicleLogs);
+
+  const solutionType = vehicleLogs.length === 0 ? "LMFS" : "ODRD";
+
+  const rawLogs = _(vehicleLogs)
```

Comment:
Can we assume here that deliveryVehicle and vehicle ids are unique?

Reply:
Probably not.  I filed an issue for future cleanup: https://github.com/googlemaps/fleet-debugger/issues/66